dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -238,7 +231,10 @@ private final class RetryingCallback {
         private RetryingCallback(
                 Endpoint endpoint, Request request, Optional<SafeRuntimeException> callsiteStacktrace) {
             this.endpoint = endpoint;
-            this.request = request;
+            this.request = Request.builder()
+                    .from(request)
+                    .body(request.body().map(ConsumptionTrackingRequestBody::new))
+                    .build();
```

Comment:
Let's factor this out into a `private static Request trackNonRepeatableBodyConsumption(Request input) {` method (no strong opinion on the name, this is just an example).
If the request body is repeatable or not present, we should pass along the request as is without wrapping/copying.

Reply:
> If the request body is repeatable or not present, we should pass along the request as is without wrapping/copying.

Nice. This is better: only using the tracking logic if needed.